### PR TITLE
Gaussian Splatting: add a gaussian splatting mesh and material

### PR DESCRIPTION
**Pull request by Popov72**
Replaces #14585

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14711/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14711/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14711/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14711/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14711/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14711/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14711/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14711/merge#BCU1XR#0

File: packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+            // udpate on view changed
+            else if (e.data.view) {
+                viewProj = e.data.view;
+                const dot = lastProj[2] * viewProj[2] + lastProj[6] * viewProj[6] + lastProj[10] * viewProj[10];
+                if (!positions || Math.abs(dot - 1) < 0.01) {
+                    return;
+                }
+
+                // Sort
+                const depthMix = new BigInt64Array(vertexCount);
```
Comment (sebavan) [inline packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts:143]: Could we keep both (working arrays) around ?

Comment (sebavan) [reply packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts:143]: I mean the three of them :-) depthMix indices and floatmix ?

**Review by sebavan — COMMENTED** [review]
Looks great to me !!!

Just wondering if the extra folders are needed as we only have one class in each of them ? I am "opinionless" but just wondering :-)

File: packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts (hunk 1 above)
Comment (Popov72) [reply packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts:143]: I tried some things but couldn't make it work, the problem is that `depthMix` would have to be passed back and forth between the worker and the main thread if we want to share it. I think we will have to discuss about it.

**Comment by Popov72** [comment]
> Just wondering if the extra folders are needed as we only have one class in each of them ? I am "opinionless" but just wondering :-)

Same, but as @CedricGuillemet created a new directory in the initial PR, I also created one :)

File: packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts [hunk 2]
```diff
@@ -74,16 +82,45 @@ export class GaussianSplattingMesh extends Mesh {
...
         }
 
         return super.render(subMesh, enableAlphaMode, effectiveMeshReplacement);
     }
 
+    /**
+     * @internal
+     */
+    public _bind(subMesh: SubMesh, effect: Effect, fillMode: number, allowInstancedRendering = true): Mesh {
+        if (this._covariancesATexture) {
```
Comment (sebavan) [inline packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts:110]: last question about this part :-)

I wonder if the material should pull the data instead in its bind function ? It makes creating other GSPlat mat or shader mat equivalent simpler I guess ?

**Review by sebavan — APPROVED** [review]

**Pull request merged**